code/modules/jobs/job_exp.dm

```diff
@@ -0,0 +1,286 @@
+	for(var/category in SSjob.name_occupations)
+		if(play_records[category])
+			exp_data[category] = text2num(play_records[category])
+		else
+			exp_data[category] = 0
+	for(var/category in GLOB.exp_specialmap)
+		if(play_records[category])
+			exp_data[category] = text2num(play_records[category])
+		else
+			exp_data[category] = 0
+	if(prefs.db_flags & DB_FLAG_EXEMPT)
+		return_text += "<LI>Exempt (all jobs auto-unlocked)</LI>"
+
+	for(var/dep in exp_data)
+		if(exp_data[dep] > 0)
+			if(exp_data[EXP_TYPE_LIVING] > 0)
+				var/percentage = num2text(round(exp_data[dep]/exp_data[EXP_TYPE_LIVING]*100))
+				return_text += "<LI>[dep] [get_exp_format(exp_data[dep])] ([percentage]%)</LI>"
+			else
+				return_text += "<LI>[dep] [get_exp_format(exp_data[dep])] </LI>"
+	if(config.use_exp_restrictions_admin_bypass && check_rights(R_ADMIN, 0, mob))
+		return_text += "<LI>Admin (all jobs auto-unlocked)</LI>"
+	return_text += "</UL>"
+	var/list/jobs_locked = list()
+	var/list/jobs_unlocked = list()
+	for(var/datum/job/job in SSjob.occupations)
+		if(job.exp_requirements && job.exp_type)
+			if(!job_is_xp_locked(job.title))
+				continue
+			else if(!job.required_playtime_remaining(mob.client))
+				jobs_unlocked += job.title
+			else
+				var/xp_req = job.get_exp_req_amount()
+				jobs_locked += "[job.title] [get_exp_format(text2num(calc_exp_type(job.get_exp_req_type())))] / [get_exp_format(xp_req)] as [job.get_exp_req_type()])"
+	if(jobs_unlocked.len)
+		return_text += "<BR><BR>Jobs Unlocked:<UL><LI>"
+		return_text += jobs_unlocked.Join("</LI><LI>")
+		return_text += "</LI></UL>"
+	if(jobs_locked.len)
+		return_text += "<BR><BR>Jobs Not Unlocked:<UL><LI>"
+		return_text += jobs_locked.Join("</LI><LI>")
+		return_text += "</LI></UL>"
+	return return_text
+
+
+/client/proc/get_exp_living()
+	if(!prefs.exp)
+		return "No data"
+	var/exp_living = text2num(prefs.exp[EXP_TYPE_LIVING])
+	return get_exp_format(exp_living)
+
+/proc/get_exp_format(expnum)
+	if(expnum > 60)
+		return num2text(round(expnum / 60)) + "h"
+	else if(expnum > 0)
+		return num2text(expnum) + "m"
+	else
+		return "0h"
+
+/datum/controller/subsystem/blackbox/proc/update_exp(mins, ann = FALSE)
+	if(!SSdbcore.Connect())
+		return -1
+	for(var/client/L in GLOB.clients)
+		if(L.is_afk())
+			continue
+		addtimer(CALLBACK(L,/client/proc/update_exp_list,mins,ann),10)
+
+/datum/controller/subsystem/blackbox/proc/update_exp_db()
+	SSdbcore.MassInsert(format_table_name("role_time"),GLOB.exp_to_update,TRUE)
+	LAZYCLEARLIST(GLOB.exp_to_update)
+
+//resets a client's exp to what was in the db.
+/client/proc/set_exp_from_db()
+	if(!config.use_exp_tracking)
+		return
+	if(!SSdbcore.Connect())
+		return -1
+	var/datum/DBQuery/exp_read = SSdbcore.NewQuery("SELECT job, minutes FROM [format_table_name("role_time")] WHERE ckey = '[sanitizeSQL(ckey)]'")
+	if(!exp_read.Execute())
+		var/err = exp_read.ErrorMsg()
+		log_sql("SQL ERROR during exp_update_client read. Error : \[[err]\]\n")
```

Comment:
When I reverted it was.
There was no message being logged to the game log with execute